examples/features/stats_monitoring/statshandler/handler.go

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package statshandler is an example pkg to illustrate the use of the stats handler.
+package statshandler
+
+import (
+	"context"
+	"log"
+	"net"
+	"path/filepath"
+
+	"google.golang.org/grpc/stats"
+)
+
+// Handler implements [stats.Handler](https://pkg.go.dev/google.golang.org/grpc/stats#Handler) interface.
+type Handler struct{}
+
+type connStatCtxKey struct{}
+
+// TagConn can attach some information to the given context.
+// The context used in HandleConn for this connection will be derived from the context returned.
+// In gRPC client:
+// The context used in HandleRPC for RPCs on this connection will NOT be derived from the context returned.
```

Comment:
Maybe explain that the context used for RPCs is the _user's_ RPC, which is why this is the case.